File: fizz_buzz.js
const fizzBuzz = (arr) => {
    return arr.map((num) => {
        if ( num % 3 === 0 && num % 5 === 0 ) {
            return "FizzBuzz"; 
        } else if ( num % 3 === 0 ){
            return "Fizz"; 
        } else if ( num % 5 === 0 ){
            return "Buzz"; 
        } else {
            return num; 
        }
    })
}

console.log(fizzBuzz([1,2,3,4,5,6,7,8,9,10,11,12,13,14,15])); 

File: 08. classic algo questions/fizz_buzz.rb
class FizzBuzz

    def nums (max_num)
        (1..max_num).map {|num| output(num)}
    end

    def output (num)
        str = ''
        if num % 3 === 0 then str += 'fizz' end
        if num % 5 === 0 then str += 'buzz' end
        if str.empty? then return num end 
        return str
    end

end

p FizzBuzz.new.nums(30)